code/07-thresholding/RootMassImproved.py

```diff
@@ -29,25 +31,23 @@
 
 # WRITE CODE HERE
 # use the mask you just created to remove the circle and label from the
-# blur image, saving the result back in the blur variable
+# blur image
 
 
 # perform adaptive thresholding to produce a binary image
-(t, binary) = cv2.threshold(src = blur, 
-    thresh = 0, 
-    maxval = 255, 
-    type = cv2.THRESH_BINARY + cv2.THRESH_OTSU)
+t = skimage.filters.thresh_otsu(blur)
+binary = blur > t
 
 # save binary image; first find extension beginning
 dot = filename.index(".")
 binaryFileName = filename[:dot] + "-binary" + filename[dot:]
-cv2.imwrite(filename = binaryFileName, img = binary)
+skimage.io.imwsave(filename = binaryFileName, img = binary)
```

Comment:
is this supposed to be `fname` instead of `filename`?

Reply:
plus there is a typo...

```suggestion
skimage.io.imsave(fname=binaryFileName, img=binary)
```